tests/net/context/src/main.c

```diff
@@ -998,43 +1034,48 @@ static void test_init(void)
 
 	/* The semaphore is there to wait the data to be received. */
 	k_sem_init(&wait_data, 0, UINT_MAX);
+
+	return NULL;
 }
 
-/*test case main entry*/
-void test_main(void)
+static void net_ctx_common_tests(void)
 {
-	ztest_test_suite(test_context,
-			 ztest_unit_test(test_init),
-			 ztest_unit_test(test_net_ctx_get_fail),
-			 ztest_unit_test(test_net_ctx_get_all),
-			 ztest_unit_test(test_net_ctx_get_success),
-			 ztest_unit_test(test_net_ctx_create),
-			 ztest_unit_test(test_net_ctx_bind_fail),
-			 ztest_unit_test(test_net_ctx_bind_uni_success_v6),
-			 ztest_unit_test(test_net_ctx_bind_uni_success_v4),
-			 ztest_unit_test(test_net_ctx_bind_mcast_success),
-			 ztest_unit_test(test_net_ctx_listen_v6),
-			 ztest_unit_test(test_net_ctx_listen_v4),
-			 ztest_unit_test(test_net_ctx_connect_v6),
-			 ztest_unit_test(test_net_ctx_connect_v4),
-			 ztest_unit_test(test_net_ctx_accept_v6),
-			 ztest_unit_test(test_net_ctx_accept_v4),
-			 ztest_unit_test(test_net_ctx_send_v6),
-			 ztest_unit_test(test_net_ctx_send_v4),
-			 ztest_unit_test(test_net_ctx_sendto_v6),
-			 ztest_unit_test(test_net_ctx_sendto_v4),
-			 ztest_unit_test(test_net_ctx_recv_v6),
-			 ztest_unit_test(test_net_ctx_recv_v4),
-			 ztest_unit_test(test_net_ctx_recv_v6_fail),
-			 ztest_unit_test(test_net_ctx_recv_v4_fail),
-			 ztest_unit_test(test_net_ctx_recv_v6_again),
-			 ztest_unit_test(test_net_ctx_recv_v4_again),
-			 ztest_unit_test(test_net_ctx_recv_v6_reconfig),
-			 ztest_unit_test(test_net_ctx_recv_v4_reconfig),
-			 ztest_unit_test(test_net_ctx_recv_v6_timeout),
-			 ztest_unit_test(test_net_ctx_recv_v4_timeout),
-			 ztest_unit_test(test_net_ctx_recv_v6_timeout_forever),
-			 ztest_unit_test(test_net_ctx_recv_v4_timeout_forever),
-			 ztest_unit_test(test_net_ctx_put));
-	ztest_run_test_suite(test_context);
+	net_ctx_bind_fail();
+	net_ctx_bind_uni_success_v6();
+	net_ctx_bind_uni_success_v4();
+	net_ctx_bind_mcast_success();
+	net_ctx_listen_v6();
+	net_ctx_listen_v4();
+	net_ctx_connect_v6();
+	net_ctx_connect_v4();
+	net_ctx_accept_v6();
+	net_ctx_accept_v4();
 }
+
+static void net_ctx_before(void *dummy)
+{
+	ARG_UNUSED(dummy);
+	net_ctx_create();
+	net_ctx_common_tests();
```

Comment:
So we absolutely have to run all the tests in `net_ctx_common_tests()` before executing the other ZTESTs?

A suggestion: If these tests are incredibly order-dependent then they're essentially one large test that could really just be all wrapped into a single ZTEST.

Reply:
Well, it seems there is another rule to split the test case as possible as we can, so that the bug can be reported accurately.
In the current PR, net_ctx_common_tests() will be run many times. I think it is OK, it will not add much time to the entire test suite

Reply:
Yes this is the case, but we if we have tests that depend on each other based on the order of execution it really is just a single test.